From 45867ae14693319e70831a05a789481040181950 Mon Sep 17 00:00:00 2001
From: Alexey Romanov <romanov.alexey2000@gmail.com>
Date: Wed, 11 Mar 2026 13:05:17 +0300
Subject: [PATCH] lib: utils: add ability to run post overlay hooks

This is useful if you want to modify the overlay "on the fly" while
kayak is running. For example, depending on environment variables.

Signed-off-by: Alexey Romanov <romanov.alexey2000@gmail.com>
---
 lib/utils.sh | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/lib/utils.sh b/lib/utils.sh
index 31cdf90..4d9cc98 100644
--- a/lib/utils.sh
+++ b/lib/utils.sh
@@ -149,6 +149,7 @@ function logadm_rsyslog {
 function apply_custom_overlay {
     typeset custom_overlay="$1"
     typeset workdir="$2"
+    typeset hooks_dir=".overlay-hooks"
 
     [ -z "$custom_overlay" ] && return 0
     [ ! -d "$custom_overlay" ] && return 0
@@ -170,6 +171,19 @@ function apply_custom_overlay {
 
     popd >/dev/null
     echo " Applied $file_count file(s) from $dir_count directories"
+
+    if [ -d "$workdir/$hooks_dir" ]; then
+        echo " --- running post-overlay hooks"
+        for hook in "$workdir/$hooks_dir"/*.sh; do
+            [ ! -f "$hook" ] && continue
+            echo " Running hook: $(basename $hook)"
+            bash "$hook" "$workdir" || {
+                echo " WARNING: Hook $(basename $hook) failed"
+            }
+        done
+        rm -rf "$workdir/$hooks_dir"
+        echo " Post-overlay hooks completed"
+    fi
 }
 
 # Vim hints
-- 
2.47.1.windows.1

